File: front/now_v2/lib/core/domain/models/spot.dart
import "package:json_annotation/json_annotation.dart";

part 'spot.g.dart';

@JsonSerializable()
class Locations {
  final List<Spot> places;

  Locations({required this.places});

  factory Locations.fromJson(Map<String, dynamic> json) =>
      _$LocationsFromJson(json);
}

@JsonSerializable()
class Spot {
  final EventInfo eventInfo;
  final PlaceInfo placeInfo;

  Spot({
    required this.eventInfo,
    required this.placeInfo,
  });

  factory Spot.fromJson(Map<String, dynamic> json) => _$SpotFromJson(json);
}

@JsonSerializable()
class EventInfo {
  final String name;
  final String id;
  final String eventType;
  final String emoji;

  EventInfo({
    required this.name,
    required this.id,
    required this.eventType,
    required this.emoji,
  });

  factory EventInfo.fromJson(Map<String, dynamic> json) =>
      _$EventInfoFromJson(json);
}

@JsonSerializable()
class PlaceInfo {
  final String mapProviderId;
  final double lat;
  final double lon;

  PlaceInfo({
    required this.mapProviderId,
    required this.lat,
    required this.lon,
  });

  factory PlaceInfo.fromJson(Map<String, dynamic> json) =>
      _$PlaceInfoFromJson(json);
}

File: services/filter/internal/core/domain/spot.go
package domain

type Locations struct {
	Places []Spot `json:"places"`
}

type Spot struct {
	DateInfo   Date   `json:"dateInfo,omitempty"`
	EventInfo  Event  `json:"eventInfo,omitempty"`
	HostInfo   Person `json:"hostInfo,omitempty"`
	PlaceInfo  Place  `json:"placeInfo,omitempty"`
	TopicsInfo Topic  `json:"topicInfo,omitempty"`
}

type Date struct {
	DateTime                      string   `json:"dateTime,omitempty"`
	Type                          SpotType `json:"type,omitempty"`
	Id                            string   `json:"id,omitempty"`
	DurationApproximatedInSeconds int64    `json:"durationApproximated,omitempty"`
	StartTime                     string   `json:"startTime,omitempty"`
}
type Event struct {
	Name           string `json:"name,omitempty"`
	UUID           string `json:"id,omitempty"`          // Only fields with Cappital letter are exported / Visibles to the outside
	Description    string `json:"description,omitempty"` // So as Json use it, we need to make ut public.
	MaximunCapacty int64  `json:"maximunCapacty,omitempty"`
	Emoji          string `json:"emoji,omitempty"`
}

type Person struct {
	Id   string `json:"id,omitempty"`
	Name string `json:"name,omitempty"`
}

type Place struct {
	Name          string  `json:"name,omitempty"`
	Lat           float64 `json:"lat,omitempty"` // those nas are tags!
	Lon           float64 `json:"lon,omitempty"`
	MapProviderId string  `json:"mapProviderId,omitempty"`
}

type Topic struct {
	// deprecated
	Name            string   `json:"name,omitempty"`
	PrincipalTopic  string   `json:"principalTopic,omitempty"`
	SecondaryTopics []string `json:"secondaryTopics,omitempty"`
}
